File: Data Structures & Algorithms/jump-game/submission-1.go
func canJump(nums []int) bool {
    last := 0

    for i := 0; i < len(nums); i++ {
        if i > last {
            return false
        }

        last = max(last, i+nums[i])
        if last >= len(nums)-1 {
            return true
        }
    }

    return true
}

func max(a, b int) int {
    if a < b {
        return b
    }
    return a
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        let res = 0;
        const set = new Set(nums);

        for (let num of nums) {
            let t = num;
            while (set.has(t+1)) t++;
            res = Math.max(res, t - num + 1);
        }

        return res;
    }
}
